aptos-move/aptos-gas-profiling/src/profiler.rs

```diff
@@ -332,20 +381,27 @@ where
         //
         // Later when it realizes the function is native, it will transform the original frame
         // into a native-specific event that does not contain recursive structures.
-        let cur = self.frames.pop().expect("frame must exist");
-        let (module_id, name, ty_args) = match cur.name {
+        let frame = self.frames.pop().expect("frame must exist");
+
+        // Add native gas accumulated per frame.
+        let (mut cost, res) =
+            self.delegate_charge(|base| base.charge_native_function(amount, ret_vals));
+        cost += frame.native_gas;
```

Comment:
Ah indeed we have an interesting problem here: we no longer have a single charge at the end of a call to a native function. Instead it gets split into smaller charges.
1. Have you tried gas profiling with this new implementation? Need to make sure it still records the charges correctly.
2. I'm fine with the current method, but wonder if there is a conceptually cleaner approach, which is to record the balances before and after native function execution and compute the difference, so that we do not maintain a separate counter, which could be the source for inconsistency. To do this, we may need to introduce new callbacks to the `GasMeter` trait.

Reply:
Yes, I actually found out about this after running tests `gas::test_gas` which compares profiler and meter. This is why `charge_native_function` cannot go away even in new flow even though charges are 0 at this point - I need to know end point so gas can be added. We already have `charge_native_function_before_execution`, I can use this if you think it is cleaner? (currently it is a no-op)

Reply:
Will also run profiler manually to double check 👍